moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -456,9 +464,27 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
   KinematicState current_state(num_joints), target_state(num_joints);
   target_state.joint_names = joint_names;
 
-  // Copy current kinematic data from RobotState.
-  robot_state->copyJointGroupPositions(joint_model_group, current_state.positions);
-  robot_state->copyJointGroupVelocities(joint_model_group, current_state.velocities);
+  // remove commands not yet committed
+  auto cur_time = node_->now();
+  while (!committed_commands_.empty() &&
+         committed_commands_.back().time >
+             (cur_time + rclcpp::Duration::from_seconds(servo_params_.max_expected_latency)))
+  {
+    committed_commands_.pop_back();
+  }
+  if (!committed_commands_.empty())
+  {
+    current_state = committed_commands_.back();
+  }
+  else
+  {
+    // Copy current kinematic data from RobotState.
+    robot_state->copyJointGroupPositions(joint_model_group, current_state.positions);
+    robot_state->copyJointGroupVelocities(joint_model_group, current_state.velocities);
+    current_state.joint_names = joint_names;
+    current_state.time = cur_time;
+    committed_commands_.push_back(current_state);
+  }
```

Comment:
Can you please recreate the graphs using a `committed_commands_` of size 3 always? I think the rolling window is not necessary, as mentioned before.

In a nutshell, `committed_commands_` will be:

```
committed_commands_.clear();
...
committed_commands_.push_back(current_state);
...
committed_commands_.push_back(target_state);
...
committed_commands_.push_back(end_state);
```

Curious to see how these graphs look after that change.

![graphs](https://github.com/ros-planning/moveit2/assets/11284393/12a55acd-6dfe-43b6-a1f2-88c773790899)

Reply:
Also, this won't be resizing all the time --> better for realtime performance.

Reply:
>  Can you please recreate the graphs using a committed_commands_ of size 3 always? 

It makes the motion a little smoother than before, but there is still substantial noise. 

![Screenshot from 2023-12-11 17-06-15](https://github.com/ros-planning/moveit2/assets/28868750/7a70449f-4a48-4175-884d-816c259d9c30)
![Screenshot from 2023-12-11 17-05-31](https://github.com/ros-planning/moveit2/assets/28868750/c052627f-5406-4804-94fa-76cff2e5b7c6)